Modules/Scripted/VPAWVisualize/VPAWVisualize.py

```diff
@@ -322,24 +284,44 @@ def onApplyButton(self):
         ):
 
             # Compute output
-            self.logic.process(
-                self.ui.inputSelector.currentNode(),
-                self.ui.outputSelector.currentNode(),
-                self.ui.imageThresholdSliderWidget.value,
-                self.ui.invertOutputCheckBox.checked,
+            list_of_files = self.logic.process(
+                self.ui.DataDirectory.currentPath, self.ui.PatientPrefix.text
             )
-
-            # Compute inverted output (if needed)
-            if self.ui.invertedOutputSelector.currentNode():
-                # If additional output volume is selected then result with inverted
-                # threshold is written there
-                self.logic.process(
-                    self.ui.inputSelector.currentNode(),
-                    self.ui.invertedOutputSelector.currentNode(),
-                    self.ui.imageThresholdSliderWidget.value,
-                    not self.ui.invertOutputCheckBox.checked,
-                    showResult=False,
-                )
+            # Display output
+            self.updateOutputWidgets(list_of_files)
+
+    def clearOutputWidgets(self):
+        outputsFormLayout = self.ui.outputsCollapsibleButton.layout()
+        # Clear away the current output widgets
+        for row_number in reversed(range(outputsFormLayout.rowCount())):
+            # !!! Do we need to delete (recursively) the widgets themselves before we
+            # !!! delete the row from the layout?  Or, is Python garbage collection
+            # !!! sufficient?
+            outputsFormLayout.removeRow(row_number)
+
+    def updateOutputWidgets(self, list_of_files):
+        self.clearOutputWidgets()
+
+        outputsFormLayout = self.ui.outputsCollapsibleButton.layout()
+        for file in list_of_files:
+            try:
+                # Collapsible contains Layout contains Label contains Pixmap
+                fileCollapsible = ctk.ctkCollapsibleButton()
+                fileCollapsible.text = file
+                fileCollapsible.collapsed = True
+                fileLayout = qt.QFormLayout(fileCollapsible)
+                fileLabel = qt.QLabel()
+                fileLabel.setText(file)
+                filePixmap = qt.QPixmap()
+                if not filePixmap.load(file):
+                    raise RuntimeError(f"Could not load pixmap from file {repr(file)}")
+                pixmap_width = max(100, self.layout.geometry().width())
+
+                fileLabel.setPixmap(filePixmap.scaledToWidth(pixmap_width))
+                fileLayout.addRow(fileLabel)
+                outputsFormLayout.addRow(fileCollapsible)
+            except Exception:
```

Comment:
This `except` swallows the above `RuntimeError` and any other possible errors from above, not allowing us to see what caused the error. Consider making it

```py
except Exception as e:
  print(f"Could not show file {repr(file)}")
  raise e
```

which is how I had to edit this to see what was wrong in my case.

Reply:
How does this sound?  I'll put in logic so that any exception that is caught is followed by `raise e` if and only if the file is an image type of some sort.  If the file is not an image, I'll still do the print statement, to alert the user that a non-image file was found and is being ignored, but I won't follow with `raise e`.

Reply:
That sounds good :+1:

Reply:
We are no longer using a `try` block so the issue of whether to re-raise an exception is now moot.